### PR TITLE
fix assignment amount calculations

### PR DESCRIPTION
**Pull request by s4lly**
fix line item assignment amount calculations

## Summary by CodeRabbit

* **Bug Fixes**
  * More accurate monetary summing/rounding to prevent floating‑point errors and ensure cent-level integrity.
  * Corrected tax and tip handling for assigned items; UI now shows true total assigned amount and updated progress visuals.
  * Improved per-person tax calculation and handling of assigned vs. unassigned amounts.

* **New Features**
  * New predicates to determine line‑item and assignment states, improving split and assignment behavior.

* **Tests**
  * Extensive tests for summing, tax rules, assignment predicates, and numerous edge cases.

### CODE REVIEW

File: frontend/src/components/Receipt/ReceiptAnalysisDisplay.tsx [hunk 1]
```diff
@@ -110,25 +111,37 @@ const ReceiptAnalysisDisplay = ({
...
+    totalAssignedAmount += totalPreTaxAssignedAmount * getTaxRate(receipt_data);
+  }
+
+  // Add tip and gratuity (split among people if any assignments made)
+  if (people.length > 0 && totalPreTaxAssignedAmount > 0) {
+    totalAssignedAmount +=
+      (receipt_data.tip ?? 0) + (receipt_data.gratuity ?? 0);
+  }
+
+  const unassignedAmount = Math.max(0, receiptTotal - totalAssignedAmount);
```
[the remark this answers is not in the document]
Comment (s4lly) [reply frontend/src/components/Receipt/ReceiptAnalysisDisplay.tsx]: create an issue for this. i would also consider creating a utility function for converting to cents.
